### PR TITLE
Assume 443 on empty port. Fixes #9

### LINKED ISSUE #9: Fails to connect to port 443
The JS URL parser has a fun gotcha:

Screenshot 2023-09-12 at 16 17 06

And it's what is used to parse the connect `SocketAddress`: https://github.com/Ethan-Arrowood/socket/blob/main/src/index.ts#L17. So `connect("google.com:443")` fails with:

``` 
Got error:  RangeError [ERR_SOCKET_BAD_PORT]: Port should be >= 0 and < 65536. Received type number (NaN).
    at new NodeError (node:internal/errors:400:5)
    at validatePort (node:internal/validators:390:11)
    at lookupAndConnect (node:net:1220:5)
    at Socket.connect (node:net:1186:5)
    at Object.connect (node:net:241:17)
    at new Socket (/Users/dominik/repos/playground/node-sockets/node_modules/@arrowood.dev/socket/dist/index.js:55:50)
    at connect (/Users/dominik/repos/playground/node-sockets/node_modules/@arrowood.dev/socket/dist/index.js:20:12)
    at file:///Users/dominik/repos/playground/node-sockets/test.js:17:28
    at ModuleJob.run (node:internal/modules/esm/module_job:194:25) {
  code: 'ERR_SOCKET_BAD_PORT'
}
```

I actually ran into the same thing when implementing sockets for Workers (https://github.com/cloudflare/workerd/pull/778/commits/f0080832e776300f162855215d547cf0e67bc7e4)

### PR DESCRIPTION
**Pull request by dom96**
Tested using:

```
pnpm test
```

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -10,11 +10,11 @@ export function connect(
   options?: SocketOptions,
 ): Socket {
   if (typeof address === 'string') {
-    const url = new URL(address);
+    const url = new URL("https://" + address);
```
Comment (Ethan-Arrowood) [inline src/index.ts:13]: I think we need a little more here. What if the `address` is something like `tcp://0.0.0.0:00`? We wouldn't want that prepended by `https`

File: src/index.ts [hunk 2]
```diff
@@ -10,11 +10,11 @@ export function connect(
...
     // eslint-disable-next-line no-param-reassign -- there is no harm reassigning to this param
     address = {
       hostname: url.hostname,
-      port: parseInt(url.port),
+      port: parseInt(url.port == "" ? "443" : url.port),
```
Comment (Ethan-Arrowood) [inline src/index.ts:17]: With some additional unit testing, I think this change is sufficient to fix #9

**Review by Ethan-Arrowood — CHANGES_REQUESTED** [review]

File: src/index.ts (hunk 1 above)
Comment (dom96) [reply src/index.ts:13]: This is something we should clarify in the spec: protocol shouldn't be supported in the `address`. It should always be "hostname:port".

Comment (dom96) [reply src/index.ts:13]: Tracking here https://github.com/wintercg/proposal-sockets-api/issues/13

**Comment by dom96** [comment]
Added test and updated existing tests using `tcp://`.

File: package.json [hunk 3]
```diff
@@ -40,5 +40,8 @@
   "packageManager": "pnpm@8.6.12",
   "engines": {
     "node": ">=18"
+  },
+  "dependencies": {
+    "pnpm": "8.8.0"
```
Comment (Ethan-Arrowood) [inline package.json:45]: remove this

**Comment by Ethan-Arrowood** [comment]
lmao GitHub Actions prohibits listening on port 443 https://github.com/Ethan-Arrowood/socket/actions/runs/6388518205/job/17338481439?pr=11#step:11:33 

this works fine locally. but fails in CI. I don't love relying on a public endpoint in tests, but I guess we have to in this case.

**Review by Ethan-Arrowood — APPROVED** [review]

**Pull request merged**